src/scheduler/sched_ifl_wrappers.cpp

```diff
@@ -153,7 +177,11 @@ send_attr_updates(int virtual_sd, resource_resv *resresv, struct attrl *pattr)
 preempt_job_info *
 send_preempt_jobs(int virtual_sd, char **preempt_jobs_list)
 {
-    return pbs_preempt_jobs(virtual_sd, preempt_jobs_list);
+	preempt_job_info *ret;
+
+    ret = pbs_preempt_jobs(virtual_sd, preempt_jobs_list);
+
+	return static_cast<preempt_job_info *>(handle_part_tolerance(ret));
```

Comment:
This can potentially become a leak if upon a sigpipe this function returns 0 and does not free up responses in "ret" variable.
It can also happen in all other stat functions below.